pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -455,6 +459,45 @@ public void uploadSegmentAsMultiPartV2(FormDataMultiPart multiPart,
     }
   }
 
+  @POST
+  @Path("segments/{tableName}/startBatchUpload")
+  @Produces(MediaType.APPLICATION_JSON)
+  @ApiOperation(value = "Start the batch upload", notes = "Start the batch upload")
+  public Response startBatchUpload(
+      @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
+      @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr, String body)
```

Comment:
You can directly put `StartBatchUploadRequest` as the parameter
```suggestion
      @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr, StartBatchUploadRequest request)
```

Reply:
fixed